src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -1075,15 +1124,23 @@ end = struct
         sys.gates <- Compiled (md5_digest, rust_gates) ;
 
         (* return the gates *)
-        rust_gates
+        (rust_gates, fixed_lookup_tables, runtime_tables_cfg)
 
   (** Calls [finalize_and_get_gates] and ignores the result. *)
-  let finalize t = ignore (finalize_and_get_gates t : Gates.t)
+  let finalize t =
+    ignore
+      ( finalize_and_get_gates t
+        : Gates.t
+          * Fp.t Kimchi_types.lookup_table array
+          * Fp.t Kimchi_types.runtime_table_cfg array )
 
-  let num_constraints sys = finalize_and_get_gates sys |> Gates.len
+  let num_constraints sys =
+    let gates, _, _ = finalize_and_get_gates sys in
+    Gates.len gates
 
   let to_json (sys : t) : string =
-    let gates = finalize_and_get_gates sys in
+    (* TODO: add lookup tables and runtime table cfgs *)
+    let gates, _, _ = finalize_and_get_gates sys in
```

Comment:
`Gates.to_json` should also export the lookup tables and runtime table cfgs.
It implies `caml_pasta_fp_plonk_circuit_serialize` must be changed.